shaders/OrganelleEmissive.gdshader

```diff
@@ -0,0 +1,74 @@
+// Intended for thermoplast and light-emitting vacuole
+shader_type spatial;
+render_mode depth_prepass_alpha, blend_mix, depth_draw_opaque, cull_back, diffuse_burley, specular_schlick_ggx;
+
+uniform sampler2D albedoTexture : source_color, filter_linear_mipmap, repeat_enable;
+
+uniform float dissolveValue : hint_range(0, 1);
+uniform float dissolveScale = 1.0;
+
+uniform vec4 tint : source_color = vec4(1, 1, 1, 1);
+
+uniform float roughness : hint_range(0.0, 1.0);
+uniform sampler2D roughnessTexture : hint_roughness_r, filter_linear_mipmap, repeat_enable;
+
+uniform float specular : hint_range(0.0, 1.0, 0.01);
+
+uniform sampler2D emissionTexture : source_color, hint_default_black, filter_linear_mipmap, repeat_enable;
+uniform vec4 emission : source_color = vec4(0.08, 0.08, 0.08, 1.0);
+uniform float emissionEnergy : hint_range(0.0, 100.0, 0.01) = 1;
+
+uniform float outlineWidth;
+uniform vec4 growColor : source_color;
+
+uniform float jiggleAmount = 0.0005f;
+
+const float jiggleMaxAngle = 15.f;
+const float jiggleTimeMultiplier = 0.5f;
+
+uniform sampler2D dissolveTexture : source_color;
+
+void vertex(){
+    // Offset animation
+    VERTEX.x += sin(TIME * 3.0f * jiggleTimeMultiplier) * jiggleAmount;
+    VERTEX.z += sin(TIME * 2.0f * jiggleTimeMultiplier) * jiggleAmount;
+
+    // Rotation animation
+    float angle = cos(TIME * jiggleTimeMultiplier) * PI * jiggleMaxAngle / 360.f;
+    mat4 rotation = mat4(
+        vec4(cos(angle), 0.f, -sin(angle), 0.f),
+        vec4(0.f, 1.f, 0.f, 0.f),
+        vec4(sin(angle),  0.f, cos(angle), 0.f),
+        vec4(0.f, 0.f, 0.f, 1.f)
+    );
+
+    VERTEX = (rotation * vec4(VERTEX, 1.0f)).xyz;
+}
+
+void fragment() {
+    vec2 base_uv = UV;
```

Comment:
Variables should be named in camelCase and not snake_case

Reply:
Though is there a reason why this needs to be a variable? It looks like the UV is not modified so why not just use `UV` in the places this is used?

Reply:
I saw that used somewhere (outside Thrive) and thought it might reduce the number of times it has to check the mesh. I didn't find info on that.

Reply:
I think the way it works is that godot just setups the variable once before starting the fragment shader. So I highly think it doesn't matter. And if I remember right in like GLSL or HLSL you need to specify the UV as a method parameter so it is pretty clear there that the data is copied from the mesh into local memory for the shader when setting it up so again I highly doubt it affects used memory bandwidth.